We have to fix a few things here:

Newer meson versions don't want us to only take the first of the resources arrays,
that leads to undefined references. We also have to build everything as shared lib,
since building internal static libs is pretty broken right meson right now,
see https://github.com/mesonbuild/meson/pull/3939

diff --git src/gstyle/meson.build src/gstyle/meson.build
index 4fe3287..fe4c90c 100644
--- src/gstyle/meson.build
+++ src/gstyle/meson.build
@@ -65,7 +65,7 @@ libgstyle_sources = [
   'gstyle-slidein.c',
   'gstyle-utils.c',
   'gstyle-xyz.c',
-  libgstyle_resources[0],
+  libgstyle_resources,
 ]
 
 libgstyle_deps = [
diff --git src/libeditorconfig/meson.build src/libeditorconfig/meson.build
index 6c59ace..bf07f9d 100644
--- src/libeditorconfig/meson.build
+++ src/libeditorconfig/meson.build
@@ -34,11 +34,14 @@ libeditorconfig_args = [
   '-Deditorconfig_VERSION_SUFFIX=0',
 ]
 
-libeditorconfig = static_library('editorconfig',
+libeditorconfig = shared_library('editorconfig',
   libeditorconfig_sources,
   dependencies: libeditorconfig_deps,
         c_args: libeditorconfig_args,
            pic: true,
+          install: true,
+   install_dir: pkglibdir_abs,
+ install_rpath: pkglibdir_abs,
 )
 
 libeditorconfig_dep = declare_dependency(
diff --git src/plugins/autotools/meson.build src/plugins/autotools/meson.build
index 94506d3..57f84a6 100644
--- src/plugins/autotools/meson.build
+++ src/plugins/autotools/meson.build
@@ -28,7 +28,6 @@ autotools_sources = [
   'ide-makecache-target.h',
 ]
 
-gnome_builder_plugins_sources += files(autotools_sources)       
-gnome_builder_plugins_sources += autotools_resources[0]         
-
+gnome_builder_plugins_sources += files(autotools_sources)
+gnome_builder_plugins_sources += autotools_resources
 endif
diff --git src/plugins/beautifier/meson.build src/plugins/beautifier/meson.build
index 8e67241..1f8006b 100644
--- src/plugins/beautifier/meson.build
+++ src/plugins/beautifier/meson.build
@@ -20,6 +20,6 @@ beautifier_sources = [
 ]
 
 gnome_builder_plugins_sources += files(beautifier_sources)
-gnome_builder_plugins_sources += beautifier_resources[0]
+gnome_builder_plugins_sources += beautifier_resources
 
 endif
diff --git src/plugins/c-pack/meson.build src/plugins/c-pack/meson.build
index a975601..8534bf3 100644
--- src/plugins/c-pack/meson.build
+++ src/plugins/c-pack/meson.build
@@ -18,6 +18,6 @@ c_pack_sources = [
 ]
 
 gnome_builder_plugins_sources += files(c_pack_sources)
-gnome_builder_plugins_sources += c_pack_resources[0]
+gnome_builder_plugins_sources += c_pack_resources
 
 endif
diff --git src/plugins/clang/meson.build src/plugins/clang/meson.build
index ba5c46a..d498e96 100644
--- src/plugins/clang/meson.build
+++ src/plugins/clang/meson.build
@@ -61,7 +61,7 @@ clang_includes_dep = declare_dependency(
 
 gnome_builder_plugins_deps += [clang_includes_dep]
 gnome_builder_plugins_sources += files(clang_sources)
-gnome_builder_plugins_sources += clang_resources[0]
+gnome_builder_plugins_sources += clang_resources
 
 gnome_builder_clang_deps = [ clang_dep, libide_deps, libide_dep ]
 
diff --git src/plugins/cmake/meson.build src/plugins/cmake/meson.build
index b13e998..be42d50 100644
--- src/plugins/cmake/meson.build
+++ src/plugins/cmake/meson.build
@@ -23,6 +23,6 @@ cmake_sources = [
 ]
 
 gnome_builder_plugins_sources += files(cmake_sources)
-gnome_builder_plugins_sources += cmake_resources[0]
+gnome_builder_plugins_sources += cmake_resources
 
 endif
diff --git src/plugins/code-index/meson.build src/plugins/code-index/meson.build
index 290d8cf..96ee7c1 100644
--- src/plugins/code-index/meson.build
+++ src/plugins/code-index/meson.build
@@ -23,6 +23,6 @@ code_index_sources = [
 ]
 
 gnome_builder_plugins_sources += files(code_index_sources)
-gnome_builder_plugins_sources += code_index_resources[0]
+gnome_builder_plugins_sources += code_index_resources
 
 endif
diff --git src/plugins/color-picker/meson.build src/plugins/color-picker/meson.build
index 198b4b5..31a1b53 100644
--- src/plugins/color-picker/meson.build
+++ src/plugins/color-picker/meson.build
@@ -29,7 +29,7 @@ color_picker_sources = [
 
 gnome_builder_plugins_deps += [libgstyle_dep]
 gnome_builder_plugins_sources += files(color_picker_sources)
-gnome_builder_plugins_sources += color_picker_resources[0]
+gnome_builder_plugins_sources += color_picker_resources
 
 install_data('gsettings/org.gnome.builder.plugins.color_picker_plugin.gschema.xml',
   install_dir: schema_dir,
diff --git src/plugins/command-bar/meson.build src/plugins/command-bar/meson.build
index 139844f..9f37ba5 100644
--- src/plugins/command-bar/meson.build
+++ src/plugins/command-bar/meson.build
@@ -30,6 +30,6 @@ command_bar_sources = [
 ]
 
 gnome_builder_plugins_sources += files(command_bar_sources)
-gnome_builder_plugins_sources += command_bar_resources[0]
+gnome_builder_plugins_sources += command_bar_resources
 
 endif
diff --git src/plugins/comment-code/meson.build src/plugins/comment-code/meson.build
index 3c27211..b74c262 100644
--- src/plugins/comment-code/meson.build
+++ src/plugins/comment-code/meson.build
@@ -13,6 +13,6 @@ comment_code_sources = [
 ]
 
 gnome_builder_plugins_sources += files(comment_code_sources)
-gnome_builder_plugins_sources += comment_code_resources[0]
+gnome_builder_plugins_sources += comment_code_resources
 
 endif
diff --git src/plugins/create-project/meson.build src/plugins/create-project/meson.build
index bfd6371..c1b0977 100644
--- src/plugins/create-project/meson.build
+++ src/plugins/create-project/meson.build
@@ -19,6 +19,6 @@ create_project_sources = [
 ]
 
 gnome_builder_plugins_sources += files(create_project_sources)
-gnome_builder_plugins_sources += create_project_resources[0]
+gnome_builder_plugins_sources += create_project_resources
 
 endif
diff --git src/plugins/ctags/meson.build src/plugins/ctags/meson.build
index 91c5db3..f75f80e 100644
--- src/plugins/ctags/meson.build
+++ src/plugins/ctags/meson.build
@@ -23,6 +23,6 @@ ctags_sources = [
 ]
 
 gnome_builder_plugins_sources += files(ctags_sources)
-gnome_builder_plugins_sources += ctags_resources[0]
+gnome_builder_plugins_sources += ctags_resources
 
 endif
diff --git src/plugins/devhelp/meson.build src/plugins/devhelp/meson.build
index b07e9f9..837b4b1 100644
--- src/plugins/devhelp/meson.build
+++ src/plugins/devhelp/meson.build
@@ -21,6 +21,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(devhelp_sources)
-gnome_builder_plugins_sources += devhelp_resources[0]
+gnome_builder_plugins_sources += devhelp_resources
 
 endif
diff --git src/plugins/deviced/meson.build src/plugins/deviced/meson.build
index e2393d5..5f9cba7 100644
--- src/plugins/deviced/meson.build
+++ src/plugins/deviced/meson.build
@@ -18,6 +18,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(deviced_sources)
-gnome_builder_plugins_sources += deviced_resources[0]
+gnome_builder_plugins_sources += deviced_resources
 
 endif
diff --git src/plugins/file-search/meson.build src/plugins/file-search/meson.build
index aa32604..42281e8 100644
--- src/plugins/file-search/meson.build
+++ src/plugins/file-search/meson.build
@@ -16,6 +16,6 @@ file_search_sources = [
 ]
 
 gnome_builder_plugins_sources += files(file_search_sources)
-gnome_builder_plugins_sources += file_search_resources[0]
+gnome_builder_plugins_sources += file_search_resources
 
 endif
diff --git src/plugins/flatpak/meson.build src/plugins/flatpak/meson.build
index 28982bb..2ad3239 100644
--- src/plugins/flatpak/meson.build
+++ src/plugins/flatpak/meson.build
@@ -58,6 +58,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(flatpak_sources)
-gnome_builder_plugins_sources += flatpak_resources[0]
+gnome_builder_plugins_sources += flatpak_resources
 
 endif
diff --git src/plugins/gcc/meson.build src/plugins/gcc/meson.build
index 4127bde..0404810 100644
--- src/plugins/gcc/meson.build
+++ src/plugins/gcc/meson.build
@@ -15,6 +15,6 @@ gcc_sources = [
 ]
 
 gnome_builder_plugins_sources += files(gcc_sources)
-gnome_builder_plugins_sources += gcc_resources[0]
+gnome_builder_plugins_sources += gcc_resources
 
 endif
diff --git src/plugins/gdb/meson.build src/plugins/gdb/meson.build
index 0627359..75a1b26 100644
--- src/plugins/gdb/meson.build
+++ src/plugins/gdb/meson.build
@@ -12,16 +12,19 @@ gdb_sources = [
   'gbp-gdb-plugin.c',
 ]
 
-gdbwire = static_library('gdbwire', ['gdbwire.c'],
+gdbwire = shared_library('gdbwire', ['gdbwire.c'],
   c_args: [ '-Wno-redundant-decls',
             '-Wno-switch-default',
             '-Wno-missing-declarations',
             '-Wno-uninitialized',
             '-Wno-declaration-after-statement' ],
+                      install: true,
+   install_dir: pkglibdir_abs,
+ install_rpath: pkglibdir_abs,
 )
 
 gnome_builder_plugins_sources += files(gdb_sources)
-gnome_builder_plugins_sources += gdb_resources[0]
+gnome_builder_plugins_sources += gdb_resources
 gnome_builder_plugins_link_with += gdbwire
 
 endif
diff --git src/plugins/gettext/meson.build src/plugins/gettext/meson.build
index 6bcf45e..2c5219a 100644
--- src/plugins/gettext/meson.build
+++ src/plugins/gettext/meson.build
@@ -13,6 +13,6 @@ gettext_sources = [
 ]
 
 gnome_builder_plugins_sources += files(gettext_sources)
-gnome_builder_plugins_sources += gettext_resources[0]
+gnome_builder_plugins_sources += gettext_resources
 
 endif
diff --git src/plugins/git/meson.build src/plugins/git/meson.build
index fdd9499..d5c3cdb 100644
--- src/plugins/git/meson.build
+++ src/plugins/git/meson.build
@@ -29,6 +29,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(git_sources)
-gnome_builder_plugins_sources += git_resources[0]
+gnome_builder_plugins_sources += git_resources
 
 endif
diff --git src/plugins/gnome-code-assistance/meson.build src/plugins/gnome-code-assistance/meson.build
index fee9b7a..4dc9890 100644
--- src/plugins/gnome-code-assistance/meson.build
+++ src/plugins/gnome-code-assistance/meson.build
@@ -23,7 +23,7 @@ gca_sources = [
 ]
 
 gnome_builder_plugins_sources += files(gca_sources)
-gnome_builder_plugins_sources += gca_resources[0]
+gnome_builder_plugins_sources += gca_resources
 
 install_data('org.gnome.builder.gnome-code-assistance.gschema.xml',
   install_dir: schema_dir)
diff --git src/plugins/history/meson.build src/plugins/history/meson.build
index 6c58e54..8c451bf 100644
--- src/plugins/history/meson.build
+++ src/plugins/history/meson.build
@@ -17,6 +17,6 @@ history_sources = [
 ]
 
 gnome_builder_plugins_sources += files(history_sources)
-gnome_builder_plugins_sources += history_resources[0]
+gnome_builder_plugins_sources += history_resources
 
 endif
diff --git src/plugins/html-completion/meson.build src/plugins/html-completion/meson.build
index 1b5b78f..393ef26 100644
--- src/plugins/html-completion/meson.build
+++ src/plugins/html-completion/meson.build
@@ -14,6 +14,6 @@ html_completion_sources = [
 ]
 
 gnome_builder_plugins_sources += files(html_completion_sources)
-gnome_builder_plugins_sources += html_completion_resources[0]
+gnome_builder_plugins_sources += html_completion_resources
 
 endif
diff --git src/plugins/meson/meson.build src/plugins/meson/meson.build
index 90319af..6dd6c2e 100644
--- src/plugins/meson/meson.build
+++ src/plugins/meson/meson.build
@@ -36,7 +36,7 @@ meson_sources = [
   'gbp-meson-utils.h',
 ]
 
-gnome_builder_plugins_sources += files(meson_sources)       
-gnome_builder_plugins_sources += meson_resources[0]         
+gnome_builder_plugins_sources += files(meson_sources)
+gnome_builder_plugins_sources += meson_resources
 
 endif
diff --git src/plugins/messages/meson.build src/plugins/messages/meson.build
index 96afadb..7c4c62f 100644
--- src/plugins/messages/meson.build
+++ src/plugins/messages/meson.build
@@ -13,4 +13,4 @@ messages_sources = [
 ]
 
 gnome_builder_plugins_sources += files(messages_sources)
-gnome_builder_plugins_sources += messages_resources[0]
+gnome_builder_plugins_sources += messages_resources
diff --git src/plugins/newcomers/meson.build src/plugins/newcomers/meson.build
index f461394..abb16ee 100644
--- src/plugins/newcomers/meson.build
+++ src/plugins/newcomers/meson.build
@@ -15,6 +15,6 @@ newcomers_sources = [
 ]
 
 gnome_builder_plugins_sources += files(newcomers_sources)
-gnome_builder_plugins_sources += newcomers_resources[0]
+gnome_builder_plugins_sources += newcomers_resources
 
 endif
diff --git src/plugins/notification/meson.build src/plugins/notification/meson.build
index a306847..0314c2d 100644
--- src/plugins/notification/meson.build
+++ src/plugins/notification/meson.build
@@ -13,6 +13,6 @@ notification_sources = [
 ]
 
 gnome_builder_plugins_sources += files(notification_sources)
-gnome_builder_plugins_sources += notification_resources[0]
+gnome_builder_plugins_sources += notification_resources
 
 endif
diff --git src/plugins/project-tree/meson.build src/plugins/project-tree/meson.build
index 4ca0242..e8be5c4 100644
--- src/plugins/project-tree/meson.build
+++ src/plugins/project-tree/meson.build
@@ -34,6 +34,6 @@ gnome_builder_plugins_deps += dependency('vte-2.91', version: '>=0.40.2')
 gnome_builder_plugins_args += '-DHAVE_VTE'
 
 gnome_builder_plugins_sources += files(project_tree_sources)
-gnome_builder_plugins_sources += project_tree_resources[0]
+gnome_builder_plugins_sources += project_tree_resources
 
 endif
diff --git src/plugins/python-pack/meson.build src/plugins/python-pack/meson.build
index bdddd88..3e105ea 100644
--- src/plugins/python-pack/meson.build
+++ src/plugins/python-pack/meson.build
@@ -13,6 +13,6 @@ python_pack_sources = [
 ]
 
 gnome_builder_plugins_sources += files(python_pack_sources)
-gnome_builder_plugins_sources += python_pack_resources[0]
+gnome_builder_plugins_sources += python_pack_resources
 
 endif
diff --git src/plugins/qemu/meson.build src/plugins/qemu/meson.build
index 0b50c10..259180c 100644
--- src/plugins/qemu/meson.build
+++ src/plugins/qemu/meson.build
@@ -12,6 +12,6 @@ qemu_sources = [
 ]
 
 gnome_builder_plugins_sources += files(qemu_sources)
-gnome_builder_plugins_sources += qemu_resources[0]
+gnome_builder_plugins_sources += qemu_resources
 
 endif
diff --git src/plugins/quick-highlight/meson.build src/plugins/quick-highlight/meson.build
index a22cdb4..57d744d 100644
--- src/plugins/quick-highlight/meson.build
+++ src/plugins/quick-highlight/meson.build
@@ -15,6 +15,6 @@ quick_highlight_sources = [
 ]
 
 gnome_builder_plugins_sources += files(quick_highlight_sources)
-gnome_builder_plugins_sources += quick_highlight_resources[0]
+gnome_builder_plugins_sources += quick_highlight_resources
 
 endif
diff --git src/plugins/recent/meson.build src/plugins/recent/meson.build
index e8501df..63e82cd 100644
--- src/plugins/recent/meson.build
+++ src/plugins/recent/meson.build
@@ -13,4 +13,4 @@ recent_sources = [
 ]
 
 gnome_builder_plugins_sources += files(recent_sources)
-gnome_builder_plugins_sources += recent_resources[0]
+gnome_builder_plugins_sources += recent_resources
diff --git src/plugins/retab/meson.build src/plugins/retab/meson.build
index c0b61aa..0a9dee0 100644
--- src/plugins/retab/meson.build
+++ src/plugins/retab/meson.build
@@ -13,6 +13,6 @@ retab_sources = [
 ]
 
 gnome_builder_plugins_sources += files(retab_sources)
-gnome_builder_plugins_sources += retab_resources[0]
+gnome_builder_plugins_sources += retab_resources
 
 endif
diff --git src/plugins/snippets/meson.build src/plugins/snippets/meson.build
index e6652d7..29ab296 100644
--- src/plugins/snippets/meson.build
+++ src/plugins/snippets/meson.build
@@ -15,6 +15,6 @@ snippets_sources = [
 ]
 
 gnome_builder_plugins_sources += files(snippets_sources)
-gnome_builder_plugins_sources += snippets_resources[0]
+gnome_builder_plugins_sources += snippets_resources
 
 endif
diff --git src/plugins/spellcheck/meson.build src/plugins/spellcheck/meson.build
index 4cf50e1..fce30cd 100644
--- src/plugins/spellcheck/meson.build
+++ src/plugins/spellcheck/meson.build
@@ -33,6 +33,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(spellcheck_sources)
-gnome_builder_plugins_sources += spellcheck_resources[0]
+gnome_builder_plugins_sources += spellcheck_resources
 
 endif
diff --git src/plugins/support/meson.build src/plugins/support/meson.build
index 1e1d9fb..8d28015 100644
--- src/plugins/support/meson.build
+++ src/plugins/support/meson.build
@@ -15,6 +15,6 @@ support_sources = [
 ]
 
 gnome_builder_plugins_sources += files(support_sources)
-gnome_builder_plugins_sources += support_resources[0]
+gnome_builder_plugins_sources += support_resources
 
 endif
diff --git src/plugins/symbol-tree/meson.build src/plugins/symbol-tree/meson.build
index a085ab7..a2d641c 100644
--- src/plugins/symbol-tree/meson.build
+++ src/plugins/symbol-tree/meson.build
@@ -18,6 +18,6 @@ symbol_tree_sources = [
 ]
 
 gnome_builder_plugins_sources += files(symbol_tree_sources)
-gnome_builder_plugins_sources += symbol_tree_resources[0]
+gnome_builder_plugins_sources += symbol_tree_resources
 
 endif
diff --git src/plugins/sysmon/meson.build src/plugins/sysmon/meson.build
index 7b5dce7..3bc857b 100644
--- src/plugins/sysmon/meson.build
+++ src/plugins/sysmon/meson.build
@@ -14,6 +14,6 @@ sysmon_sources = [
 ]
 
 gnome_builder_plugins_sources += files(sysmon_sources)
-gnome_builder_plugins_sources += sysmon_resources[0]
+gnome_builder_plugins_sources += sysmon_resources
 
 endif
diff --git src/plugins/sysprof/meson.build src/plugins/sysprof/meson.build
index 09c47e7..815d1c9 100644
--- src/plugins/sysprof/meson.build
+++ src/plugins/sysprof/meson.build
@@ -20,6 +20,6 @@ gnome_builder_plugins_deps += [
 ]
 
 gnome_builder_plugins_sources += files(sysprof_sources)
-gnome_builder_plugins_sources += sysprof_resources[0]
+gnome_builder_plugins_sources += sysprof_resources
 
 endif
diff --git src/plugins/sysroot/meson.build src/plugins/sysroot/meson.build
index 01c2b4d..9cd2ce6 100644
--- src/plugins/sysroot/meson.build
+++ src/plugins/sysroot/meson.build
@@ -25,6 +25,6 @@ sysroot_sources = [
 ]
 
 gnome_builder_plugins_sources += files(sysroot_sources)
-gnome_builder_plugins_sources += sysroot_resources[0]
+gnome_builder_plugins_sources += sysroot_resources
 
 endif
diff --git src/plugins/terminal/meson.build src/plugins/terminal/meson.build
index e697221..ca3d0bc 100644
--- src/plugins/terminal/meson.build
+++ src/plugins/terminal/meson.build
@@ -19,4 +19,4 @@ terminal_sources = [
 gnome_builder_plugins_deps += [libvte_dep]
 
 gnome_builder_plugins_sources += files(terminal_sources)
-gnome_builder_plugins_sources += terminal_resources[0]
+gnome_builder_plugins_sources += terminal_resources
diff --git src/plugins/todo/meson.build src/plugins/todo/meson.build
index cc76a72..44b0afc 100644
--- src/plugins/todo/meson.build
+++ src/plugins/todo/meson.build
@@ -19,6 +19,6 @@ todo_sources = [
 ]
 
 gnome_builder_plugins_sources += files(todo_sources)
-gnome_builder_plugins_sources += todo_resources[0]
+gnome_builder_plugins_sources += todo_resources
 
 endif
diff --git src/plugins/words/meson.build src/plugins/words/meson.build
index 25c5f16..eeb12d1 100644
--- src/plugins/words/meson.build
+++ src/plugins/words/meson.build
@@ -14,6 +14,6 @@ words_sources = [
 ]
 
 gnome_builder_plugins_sources += files(words_sources)
-gnome_builder_plugins_sources += words_resources[0]
+gnome_builder_plugins_sources += words_resources
 
 endif
diff --git src/plugins/xml-pack/meson.build src/plugins/xml-pack/meson.build
index 5ae9289..212c923 100644
--- src/plugins/xml-pack/meson.build
+++ src/plugins/xml-pack/meson.build
@@ -41,6 +41,6 @@ xml_pack_sources = [
 ]
 
 gnome_builder_plugins_sources += files(xml_pack_sources)
-gnome_builder_plugins_sources += xml_pack_resources[0]
+gnome_builder_plugins_sources += xml_pack_resources
 
 endif
